pingcap/tidb
# server: add Uptime status counter

### What problem does this PR solve? 

Fixes #8842

### What is changed and how it works?

Uses variable infrastructure to push `Uptime` statistic.  So you can now do this, as in MySQL:
```
tidb> show global status like 'Uptime';
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Uptime        | 10    |
+---------------+-------+
1 row in set (0.00 sec)
```

### Check List 

Tests 

 - Manual test

I would like to add a test, but I wanted to check first - since the logical way to do it is to run `show global status`, sleep a second, and then run it again.  But I don't know if:
1. Sleeping for 1 second is considered acceptable since it slows down the tests.
2. It handles the race condition successfully that sleeping is never exactly 1 second.

As an alternative, I could assert that the value is greater than zero? Feedback welcome.

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Increased code complexity

Related changes

 - Need to update the documentation (supported statuses, variables)
 - Need to be included in the release note


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pingcap/tidb/8888)
